<launch>
  <param name="/alfred_driver/broadcast_odom_tf" type="bool" value="true"/>
  <param name="/alfred_driver/mode" type="string" value="navigation" />
  <param name="robot_description" textfile="$(find alfred_core)/config/stretch.urdf" />
  <arg name="calibrated_controller_yaml_file" value="$(find alfred_core)/config/controller_calibration_head.yaml"/>

  <node
      name="joint_state_publisher"
      pkg="joint_state_publisher"
      type="joint_state_publisher" >
    <param name="rate" value="15.0"/>
    <rosparam param="source_list">
      [/alfred/joint_states]
    </rosparam>
  </node>

  <node
      name="robot_state_publisher"
      pkg="robot_state_publisher"
      type="robot_state_publisher" >
    <param name="publish_frequency" value="15.0"/>
  </node>

  <node name="alfred_driver" pkg="alfred_core" type="alfred_driver.py" output="screen">
    <param name="rate" type="double" value="15.0"/>
    <param name="timeout" type="double" value="0.5"/>
    <remap from="cmd_vel" to="/alfred/cmd_vel" />
    <remap from="joint_states" to="/alfred/joint_states" />
    <param name="controller_calibration_file" type="string" value="$(arg calibrated_controller_yaml_file)"/>
  </node>
  
</launch>